Rakefile

```diff
@@ -72,6 +72,8 @@ else
 
   Rake::JavaExtensionTask.new("puma_http11", gemspec) do |ext|
     ext.lib_dir = "lib/puma"
+    ext.source_version = '1.8'
```

Comment:
I'm going to mark as "breaking" b/c of this change (though minor!)

Reply:
happy to revert this part if it's to delay shipping the feature.
my reasoning was - theoretically this would break users using JRuby 9.1 (9.1.17.0 released on April 23 2018) which was the last to work with Java 1.7 that could still install latest Puma. 
But I do not think anyone would be upgrading smt on 9.1 and not use at least Java 8, so the "breaking" risk seemed almost non-existent given there's JRuby 9.2 already in maintenance mode and latest stable in JRuby 9.3 (all of which require Java 8+).

Reply:
I think next version release will be a major so this won't delay it. Thanks for the explanation.